assemble/conf/templates/generic_logger.xml

```diff
@@ -25,7 +25,7 @@
      <param name="MaxBackupIndex" value="10"/>
      <param name="Threshold"      value="DEBUG"/>
      <layout class="org.apache.log4j.PatternLayout">
-       <param name="ConversionPattern" value="%d{ISO8601} [%-8c{2}] %-5p: %m%n"/>
+       <param name="ConversionPattern" value="%d{ISO8601} [%-8c{2}] [pid:%6X{pid}] %-5p: %m%n"/>
```

Comment:
> However this approach was problematic because logs would not be properly rolled and aged off, which could lead to using up lots of disk space for orphaned tserver logs.

That's a good point. We would want to make sure that the defined Appender configuration works as intended for a single process (N processes each with up to X files of Z size).

HBase added some stuff to their 2.0 branch which can manage multiple regionservers. At a glance, they're using a number of track instances per host. If you run 3 RS on a host, the first started RS would have a 1 in the log name, a 2 in the second, 3 in the third. This is pushed down via a system property. I assume they have the RollingFileAppender working fine but haven't checked myself.